JustSaying/AwsTools/QueueCreation/SnsWriteConfiguration.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using Amazon.SimpleNotificationService.Model;
+
+namespace JustSaying.AwsTools.QueueCreation
+{
+    public class SnsWriteConfiguration
+    {
+        /// <summary>
+        /// Extension point allows custom error handling, including ability to specify whether exception has been explictly handled by consumer.
+        /// </summary>
+        /// <returns>Boolean to indicate whether the exception has already been handled by the consumer</returns>
+        public Func<Exception, PublishRequest, bool> OnException { get; set; }
```

Comment:
I'm thinking this should be called `HandleException` because you have to indicate if it is handled or still requires propagation.

Reply:
Also, is there a good reason to require `PublishRequest`, a lot of the extensibility tries to hide AWS specific types from the API surface, I would prefer this abstracted, of not included if not really needed, the exception contains things like endpoint etc...

Reply:
I was considering `HandleException` but decided against it simply because you may not be handling the exception. The method is invoked when an exception occurs, whether you choose to handle it is down to the consumer. Happy either way though.

Reply:
Didn't realise you were so keen on hiding the AWS API but it makes sense, main reason for including it is if the consumer wishes to log the exception the `PublishRequest` object provides more context around the request (bearing in mind the `PublishException` is not thrown if they choose to handle the exception themselves - [see here](https://github.com/justeat/JustSaying/pull/340/files#diff-f6c5b29e7f5c2decc9b2ec6ff6a449c3R79).

Happy either way on both points though!

Reply:
Cool, I vote for:
```csharp
public Func<Exception, bool> HandleException { get; set; }
```